Source/Task/ThreadPool_win32.cpp

```diff
@@ -61,10 +61,21 @@ class ThreadPoolImpl
         }
     }
 
-    void Submit() noexcept
+    // SubmitThreadpoolWork (seems to be an alias for NTDLL.TpPostWork) cannot be called during application shutdown
+    // It throws: Exception thrown at 0x00007FFD726AF220 (ntdll.dll) in CTA_PC.exe: 0xC000000D: An invalid parameter was passed to a service or function. occurred
+    // The client application that uses this crashes when we close the application using (X) close button on the upper right corner
+    //  
+    void Submit(bool isTerminating) noexcept
     {
-        m_activeCalls++;
-        SubmitThreadpoolWork(m_work);
+        if (isTerminating)
```

Comment:
Ok, I've traced the code.  This throws when the process environment block's ShutdownInProgress is set, which happens after a call to ExitProcess.  The only way user code could get called in this case is if 1) you're responding to a callback very late, or 2) you're trying to to do work inside DllMain.  For both cases, the task queue should have already been shut down by the app.  When is cat_pc.exe terminating the task queue?

Reply:
I'm gonna follow up in email with everyone on this thread, so we can get more specific than I can here.

Reply:
This happens when we're using upper right corner close (X) button. Could it be another message like WM_CLOSE or whatever we should handle and clean up properly? Although i am willing to abandon this change, since our call is part of an SDK and we don't have control on our lifecycle, i am a little concerned.

Reply:
Ok after our email discussion, I will abandon this PR.